src/processor/mapper/plan_mapper.cpp

```diff
@@ -124,12 +127,43 @@ unique_ptr<PhysicalOperator> PlanMapper::mapLogicalOperatorToPhysical(
     case LOGICAL_DROP_TABLE: {
         physicalOperator = mapLogicalDropTableToPhysical(logicalOperator.get(), mapperContext);
     } break;
+    case LOGICAL_SHORTEST_PATH: {
+        physicalOperator = mapLogicalShortestPathToPhysical(logicalOperator.get(), mapperContext);
+    } break;
     default:
         assert(false);
     }
     return physicalOperator;
 }
 
+unique_ptr<PhysicalOperator> PlanMapper::mapLogicalShortestPathToPhysical(
+    LogicalOperator* logicalOperator, MapperContext& mapperContext) {
+    auto* logicalShortestPath = (LogicalShortestPath*)logicalOperator;
+    auto prevOperator = mapLogicalOperatorToPhysical(logicalOperator->getChild(0), mapperContext);
+    auto sourceNode = logicalShortestPath->getSrcNodeExpression();
+    auto destNode = logicalShortestPath->getDestNodeExpression();
+    auto rel = logicalShortestPath->getRelExpression();
+    // assume this here for now
+    auto direction = REL_DIRECTIONS[0];
+    auto srcDataPos = mapperContext.getDataPos(sourceNode->getInternalIDPropertyName());
+    auto destDataPos = mapperContext.getDataPos(destNode->getInternalIDPropertyName());
```

Comment:
During the mapping phase, you would need to get data pos for relID, too. And pass it to the shortest path operator.